core/src/main/java/org/elasticsearch/rest/action/cat/RestThreadPoolAction.java

```diff
@@ -260,9 +150,31 @@ private Table buildTable(RestRequest req, ClusterStateResponse state, NodesInfoR
                     }
                 }
             }
-            for (String pool : SUPPORTED_NAMES) {
-                ThreadPoolStats.Stats poolStats = poolThreadStats.get(pool);
-                ThreadPool.Info poolInfo = poolThreadInfo.get(pool);
+            for (Map.Entry<String, ThreadPoolStats.Stats> entry : poolThreadStats.entrySet()) {
+
+                boolean matches = false;
+                for (final String threadPool : threadPools) {
+                    matches = entry.getKey().matches(threadPool);
```

Comment:
@nik9000 I pushed 42017c40dc2f67a52fc672dd20aacd451ecdc464 to use the simple matching, and I also pushed 731c3f71a5bd39a3092f166e7c5be6ee69c12427 to simplify the logic.